### PR TITLE
Accept metrics and categories defined in core

### PR DESCRIPTION
**Pull request by lboeman**
Imports lists of metrics and categories from core to validate against.

### CODE REVIEW

File: sfa_api/schema.py [hunk 1]
```diff
@@ -6,7 +6,9 @@
 from sfa_api import spec, ma
 from sfa_api.utils.validators import (
     TimeFormat, UserstringValidator, TimezoneValidator, TimeLimitValidator)
-from solarforecastarbiter.datamodel import ALLOWED_VARIABLES
+from solarforecastarbiter.datamodel import (
+    ALLOWED_VARIABLES, ALLOWED_CATEGORIES)
+from solarforecastarbiter.metrics.deterministic import _MAP
```
Comment (alorenzo175) [inline sfa_api/schema.py]: Can we add ALLOWED_METRICS to datamodel instead of using the "private" _MAP first?

Comment (lboeman) [reply sfa_api/schema.py]: Yeah. The `__check_metrics__` function defers validating to the metrics package, due to the differences between probabilistic and deterministic forecasts. Not sure that it's particularly useful to the data model itself, but will be here and on the dashboard. I'll add `ALLOWED_DETERMINISTIC_METRICS` and `ALLOWED_PROBABILISTIC_METRICS`.

**Comment by lboeman** [comment]
Github hasn't updated the checks section, but tests are now passing. @alorenzo175 good with you after the core update?

File: sfa_api/schema.py [hunk 2]
```diff
@@ -704,12 +706,19 @@ class ReportParameters(ma.Schema):
...
     metrics = ma.List(
         ma.String(
-            validate=validate.OneOf(ALLOWED_METRICS)
+            validate=validate.OneOf(ALLOWED_REPORT_METRICS)
         ),
         title='Metrics',
         description=('The metrics to include in the report.'),
         required=True
     )
+    categories = ma.List(
```
Comment (alorenzo175) [inline sfa_api/schema.py]: Defined twice: here and line 702

Comment (lboeman) [reply sfa_api/schema.py]: ahh yep, thanks I botched this rebasing.

**Pull request merged**